Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py

```diff
@@ -1008,7 +1041,10 @@ def network_checkpoint(self):
                 if self.pstate_for_restart is not None:
                     resdata = self.pstate_for_restart.get('state', 'engineresumedata')
             elif isinstance(self.tdef, TorrentDef):
-                resdata = self.handle.save_resume_data()
+                self.save_resume_data()
+
+                # whether it's already called or not
+                resdata = self.resume_data
```

Comment:
Yes you're right. I missed that part. It's fixed in be4370154ba96d229f3287bafd3fa3a176c5ac83 now